schema/generate_scaler_schema.go

```diff
@@ -0,0 +1,552 @@
+		metadatas = append(metadatas, metadata)
+	}
+	return metadatas
+}
+
+// getBuildScalerCalls is a function that gets the map of scaler names and the creator function names from the scalers_builder file
+func getBuildScalerCalls(fileName string) (map[string]string, error) {
+	scalerCallers := map[string]string{}
+	data, err := os.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, "", string(data), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to find the "triggerType" switch statement and get the trigger names and creator function names
+	ast.Inspect(f, func(n ast.Node) bool {
+		switch t := n.(type) {
+		case *ast.SwitchStmt:
+			s, ok := t.Tag.(*ast.Ident)
+			if !ok || s.Name != "triggerType" {
+				break
+			}
+			// retrieve the creator function name from each case statement
+			for _, casesAst := range t.Body.List {
+				c, ok := casesAst.(*ast.CaseClause)
+
+				if !ok || len(c.List) == 0 || len(c.Body) == 0 {
+					continue
+				}
+
+				basic, ok := c.List[0].(*ast.BasicLit)
+				if !ok {
+					continue
+				}
+
+				r, ok := c.Body[0].(*ast.ReturnStmt)
+				if !ok {
+					continue
+				}
+
+				if len(r.Results) == 0 {
+					continue
+				}
+				caller, ok := r.Results[0].(*ast.CallExpr)
+				if !ok {
+					continue
+				}
+
+				expr, ok := caller.Fun.(*ast.SelectorExpr)
+				if !ok {
+					continue
+				}
+
+				scalerCallers[strings.Trim(basic.Value, "\"")] = expr.Sel.Name
+			}
+		}
+		return true
+	})
+	return scalerCallers, nil
+}
+
+// getAllKedaTagedStructs is a function that gets all the structs that are tagged with `keda` from the scalers files
+func getAllKedaTagedStructs(dir string) (map[string]*ast.StructType, map[string]*ast.StructType) {
+	// loop all files in the scalers directory and get the structs that are tagged with `keda`
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	kedaScalerStructs := map[string]*ast.StructType{}
+	kedaTagStructs := map[string]*ast.StructType{}
+
+	for _, e := range entries {
+		if e.IsDir() {
+			getAllKedaTagedStructs(dir + "/" + e.Name())
+			continue
+		}
+
+		data, err := os.ReadFile(dir + "/" + e.Name())
+		if err != nil {
+			continue
+		}
+
+		fset := token.NewFileSet()
+		f, err := parser.ParseFile(fset, "", string(data), 0)
+		if err != nil {
+			continue
+		}
+
+		var scalerStructs *ast.StructType
+
+		for _, decl := range f.Decls {
+			switch v := decl.(type) {
+			case *ast.GenDecl:
+				for _, spec := range v.Specs {
+					ts, ok := spec.(*ast.TypeSpec)
+					if !ok {
+						continue
+					}
+
+					structType, ok := ts.Type.(*ast.StructType)
+					if !ok {
+						continue
+					}
+
+					hasTriggerIndex := false
+					for _, v := range structType.Fields.List {
+						// Find `triggerIndex` field and identify the struct as a scaler struct, otherwise identify the struct as an abstract substruct
+						if len(v.Names) > 0 && v.Names[0].Name == "triggerIndex" {
```

Comment:
Not all the scalers use "triggerIndex", for example in dynatrace it's "TriggerIndex". Do we need this validation?

Reply:
Then we need to document that "triggerIndex" is a mandatory field and maybe validate during the parsing or so to avoid issues. The problem is that DD scaler uses "TriggerIndex" and it's not discovered by the tool and I'd like to avoid it. 
I guess that we can modify the parser to auto-asign trigger index, ensuring that the property is present. WDYT? 
@wozniakjan  you made the parser IIRC, does it make sense to you too?